modules/platform/nuxeo-audit-storage-directory/src/main/java/org/nuxeo/audit/storage/stream/StreamAuditStorageWriter.java

```diff
@@ -113,11 +113,15 @@ protected void writeJsonEntriesToAudit(ComputationContext context) {
             if (jsonEntries.isEmpty()) {
                 return;
             }
-            log.debug("Writing {} log entries to the directory audit storage {}.", jsonEntries.size(),
-                    DirectoryAuditStorage.NAME);
             NXAuditEventsService audit = (NXAuditEventsService) Framework.getRuntime()
                                                                          .getComponent(NXAuditEventsService.NAME);
             DirectoryAuditStorage storage = (DirectoryAuditStorage) audit.getAuditStorage(DirectoryAuditStorage.NAME);
+            if (storage == null) {
```

Comment:
my understanding is that the NPE was on the above line in auidt.getAuditStorage ?
```java.lang.NullPointerException: null
	at org.nuxeo.ecm.platform.audit.service.NXAuditEventsService.getAuditStorage(NXAuditEventsService.java:214) ~[nuxeo-platform-audit-core-PR-4633-11.5-SNAPSHOT.jar:?]
	at org.nuxeo.audit.storage.stream.StreamAuditStorageWriter$AuditStorageLogWriterComputation.writeJsonEntriesToAudit(StreamAuditStorageWriter.java:122) ~[nuxeo-audit-storage-directory-PR-4633-11.5-SNAPSHOT.jar:?]
```

Reply:
Indeed, I moved the null by initializing the map in `getAuditStorage` in the `Component#start` method, so we need to check if the storage is initialized/present there now.

Reply:
perfect then